Documentation/allow_legacy_mode.md

```diff
@@ -0,0 +1,102 @@
+
+Allow-legacy helps users upgrade v0.4 etcd clusters easily, and allows your etcd cluster to have a minimal amount of downtime -- less than 1 minute for clusters storing less than 50 MB.
+
+It supports upgrading from internal v1 to internal v2 now.
+
+### Setup
+
+This mode is on if `ETCD_ALLOW_LEGACY_MODE` is set to true, or etcd is running in CoreOS system.
+
+It treats `ETCD_BINARY_DIR` as the directory for etcd binaries, which is organized in this way:
+
+```
+ETCD_BINARY_DIR
+  |
+  -- 1
+  |
+  -- 2
+```
+
+`1` is etcd with internal API v1 protocol. You should use etcd v0.4.7 here. `2` is etcd with internal API v2 protocol, which is etcd v2.x.
+
+The default value for `ETCD_BINARY_DIR` is `/usr/libexec/etcd/internal_versions/`.
+
+### Ugrading a Cluster
+
+When starting etcd with a v1 data directory and v1 flags, etcd executes the v0.4.7 binary and runs exactly the same as before.
+
+Before upgrading, you SHOULD check the health of the cluster by running:
+
+```
+$ etcdctl cluster-health
+```
+
+If the cluster and all members are healthy, you can start the upgrading process. If not, check the unhealthy machines and repair them using [admin guide](./admin_guide.md).
+
+**The critical moment**: Use the `etcdctl upgrade` command to upgrade the etcd cluster to 2.0 now:
+
+```
+$ etcdctl upgrade --old-version=1 --new-version=2 --peer-url=$PEER_URL
+```
+
+`PEER_URL` is any accessible peer url of the cluster.
+
+All peer-mode members will print out:
+
+```
+detected next internal version 2, exit after 10 seconds.
+```
+
+etcd does internal coordination for the upgrade within the cluster in this period and then exits.
+
+You need to restart them after they exit. When restarted, it upgrades the data directory, and executes etcd v2.0. If it runs on a CoreOS system, the init system can take care of restart for you.
+
+After the upgrade process, you can run the health check again to verify the upgrade. If the cluster is unhealthy or there is an unhealthy member, please refer to start [failure recovery](#failure-recovery).
+
+Standby-mode members will exit after etcd cluster is upgraded to v2.0 with the message:
+
+```
+Detect the cluster has been upgraded to internal API v2. Exit now.
+```
+
+Once restarted, standbys run in v2.0 proxy mode, which proxy user requests to the etcd cluster.
+
+### Downgrade
+
+If the upgrading fails due to disk/network issues, you still can restart the upgrading process manually. However, once you upgrade etcd to internal API v2, you CANNOT downgrade it back to internal API v1. If you want to downgrade etcd in the future, please backup your v1 data dir beforehand.
+
+### Unavailable window
+
+etcd cluster will be unavailable between the time that the upgrade command is sent and the time that the whole cluster is restarted.
```

Comment:
We should probably make it more clear that your _etcds_ restart, not the machines:

"etcd cluster will be unavailable between the time that the upgrade command is sent and the time that each etcd process is upgraded and restarted."